File: go/sprint5/K/code.go
package main

/**
Comment it before submitting
type Node struct { 
	value    int 
	left   *Node 
	right  *Node 
}
**/


func printRange(root *Node, left int, right int) {
    // Your code
    // “ヽ(´▽｀)ノ”
}

func test() {
	node1 := Node{2, nil, nil}
    node2 := Node{1, nil, &node1}
    node3 := Node{8, nil, nil}
    node4 := Node{8, nil, &node3}
    node5 := Node{9, &node4, nil}
    node6 := Node{10, &node5, nil}
    node7 := Node{5, &node2, &node6}
    printRange(&node7, 2, 8);
    // expected output: 2 5 8 8
}

File: cpp/sprint5/K/code.cpp
// ! Remove define before submitting !
#define local

#ifdef local
struct Node {  
  int value;  
  const Node* left = nullptr;  
  const Node* right = nullptr;
  Node(Node* left, Node* right, int value) : value(value), left(left), right(right) {}
};  
#endif

#ifndef local
#include "solution.h"
#endif
#include <cmath>
#include <iostream>
#include <cassert>

using namespace std;


using namespace std;

void print_range(Node* root, int l, int r) {
    // Your code
    // “ヽ(´▽｀)ノ”
}

#ifdef local
void test() {
    Node node1({nullptr, nullptr, 2});
    Node node2({nullptr, &node1, 1});
    Node node3({nullptr, nullptr, 8});
    Node node4({nullptr, &node3, 8});
    Node node5({&node4, nullptr, 9});
    Node node6({&node5, nullptr, 10});
    Node node7({&node2, &node6, 5});
    print_range(&node7, 2, 8);
    // expected output: 2 5 8 8
}

int main() {
  test();
}
#endif